src/clients/SpokePoolClient.ts

```diff
@@ -664,25 +679,29 @@ export class SpokePoolClient extends BaseAbstractClient {
     // is heavy as there is a fair bit of block number lookups that need to happen. Note this call REQUIRES that the
     // hubPoolClient is updated on the first before this call as this needed the the L1 token mapping to each L2 token.
     if (eventsToQuery.includes("FundsDeposited")) {
-      const allDeposits = [
-        ...(queryResults[eventsToQuery.indexOf("FundsDeposited")] as FundsDepositedEvent[]),
-        ...this.earlyDeposits,
-      ];
-      const { earlyDeposits = [], depositEvents = [] } = groupBy(allDeposits, (depositEvent) => {
-        if (this._isEarlyDeposit(depositEvent, currentTime)) {
-          const { args, transactionHash } = depositEvent;
-          this.logger.debug({
-            at: "SpokePoolClient#update",
-            message: "Deferring early deposit event.",
-            currentTime,
-            deposit: { args, transactionHash },
-          });
-          return "earlyDeposits";
-        } else {
-          return "depositEvents";
-        }
-      });
+      // Filter out any early v2 deposits (quoteTimestamp > HubPoolClient.currentTime). Early deposits are no longer a
+      // critical risk in v3, so don't worry about filtering those. This will reduce complexity in several places.
```

Comment:
This is a really nice observation about early deposits in v3, just confirming that this is because the relayer no longer needs to set `realizedLpFeePct` correctly, only approximate it?

Reply:
Yeah - the relayer only needs to know `realizedLpFeePct` for its internal profitability logic. Handling of "early" deposits clutters the code in a few places, so I'll advocate to remove it once `deposit()` has been removed.